From 86489acb0165aa30fdb2b7d9ae4e6114dc43e0b6 Mon Sep 17 00:00:00 2001
From: Jacky Chou <jacky_chou@aspeedtech.com>
Date: Tue, 5 May 2026 17:01:29 +0800
Subject: [PATCH 04/13] net: dwc_eth_xgmac: guard SoCFPGA compatible match

Only expose the Intel SoCFPGA compatible when the SoCFPGA glue is
enabled. This prevents non-SoCFPGA builds from depending on the
SoCFPGA-specific config data that is built from a separate object.

Signed-off-by: Jacky Chou <jacky_chou@aspeedtech.com>
Change-Id: I1b5ddfba0d0d5482fd574d9ec5f6062b11cf5956
Upstream-Status: Backport [SDK v11.03 migration]
---
 drivers/net/dwc_eth_xgmac.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/dwc_eth_xgmac.c b/drivers/net/dwc_eth_xgmac.c
index 311b57011c..d179fe287d 100644
--- a/drivers/net/dwc_eth_xgmac.c
+++ b/drivers/net/dwc_eth_xgmac.c
@@ -1199,10 +1199,12 @@ static const struct eth_ops xgmac_ops = {
 };
 
 static const struct udevice_id xgmac_ids[] = {
+#if CONFIG_IS_ENABLED(DWC_ETH_XGMAC_SOCFPGA)
 	{
 		.compatible = "intel,socfpga-dwxgmac",
 		.data = (ulong)&xgmac_socfpga_config
 	},
+#endif
 	{ }
 };
 
-- 
2.34.1

